dl1_data_handler/reader.py

```diff
@@ -536,6 +536,8 @@ def _construct_mono_example_identifiers(self):
                     dl1_tel_table.add_column(
                         np.arange(len(dl1_tel_table)), name="table_index", index=0
                     )
+                    # Unique the table to remove unwanted duplication 
+                    dl1_tel_table = unique(dl1_tel_table, keys=["obs_id", "event_id", "tel_id"])
```

Comment:
I think this should not happen silently but raise a massive warning

Reply:
Also note that `unique` sorts the output table  by `keys`. This can be an unwanted side effect.

Reply:
Yes, some warnings would be good here. Also, I think the `process-tool` should not let me write in one table two entries with equal keys (obs, event, and tel ids). Not sure if this would be easy to check?

Thanks for the heads-up for the sorting. In the past, some ordering or not ordering of astropy functions gave me some headache, but here we do not care about the order since the table index for the row is added as a column before. `astropy v7.X ` introduces some nice functionality to `keep_order` (but I think you are well aware for it ;-) )